src/backend/src/services/calendar.services.ts

```
@@ -81,4 +83,50 @@ export default class CalendarService {
 
     return eventTypeTransformer(newEventType);
   }
+
+  /**
+   * Creates a new machinery and associates it with shops.
+   *
+   * @param submitter The user submitting the request, who must be an admin.
+   * @param name The name of the machinery.
+   * @param shopMachineryData Array of shop machinery data containing shopId, quantity, and optional description.
+   * @param organization The organization for which the machinery is being created.
+   * @param description The description of the machinery (optional).
+   *
+   * @returns The created machinery object with associated shop machinery.
+   *
+   * @throws AccessDeniedAdminOnlyException If the submitter is not an admin.
+   */
+  static async createMachinery(
+    submitter: User,
+    name: string,
+    shopId: string,
+    quantity: number,
+    organization: Organization,
+    description?: string
+  ) {
+    // Check if user is admin
+    if (!(await userHasPermission(submitter.userId, organization.organizationId, isAdmin))) {
+      throw new AccessDeniedAdminOnlyException('create machinery');
+    }
```

Check if shop with that id exists to throw not found exception if it doesn't